public/components/trace_analytics/components/services/service_view.tsx

```diff
@@ -178,11 +179,20 @@ export function ServiceView(props: ServiceViewProps) {
                 name: 'View logs',
                 'data-test-subj': 'viewLogsButton',
                 onClick: () => {
-                  if (isNewNavEnabled) {
+                  // NOTE: Discover has issue with PPL Time filter, hence adding +3/-3 days to actual timestamp
+                  const startTime =
+                    dateMath
+                      .parse(props.startTime)!
+                      .subtract(3, 'days')
+                      .format(TRACE_ANALYTICS_DATE_FORMAT) ?? 'now-1y';
+                  const endTime =
+                    dateMath
+                      .parse(props.startTime, { roundUp: true })!
```

Comment:
won't this go into the future? We were thinking of different logic to append 23:59:59 to end day and 00:00:00 to startDay.

Reply:
Yes this will go in future, this is to offset Discover UTC issue. But, from what we discussed even `append 23:59:59 to end day` should go in future.